docs/configuration/jobs.rst

```diff
@@ -173,6 +173,145 @@ Note that you will have to trigger replication manually using the ``zrepl signal
        type: manual
      ...
 
+.. _job-snapshotting-hooks:
+
+Pre- and post-snapshot Hooks
+----------------------------
+
+Jobs with `periodic snapshots <job-snapshotting-spec_>`_ can run a hook script or executable before and/or after taking the snapshot,
+configured in an optional ``hooks`` section of the ``snapshotting`` configuration:
+
+::
+
+
+    jobs:
+    - type: push
+      filesystems: {
+        "<": true,
+        "tmp": false
+      }
+      snapshotting:
+        type: periodic
+        prefix: zrepl_
+        interval: 10m
+        hooks:
+        - type: script
+          pre: /path/to/hooks/pre-snapshot.sh
+          post: /path/to/hooks/zrepl-notify.sh
+          timeout: 30s
+          stop_after_error: false
+        - type: script
+          pre: /path/to/hooks/special-snapshot.sh
+          filesystems: {
+            "tank/special": true
+          }
+      ...
+
+The ``type: script`` parameter is mandatory. Future versions of Zrepl may support other hook types.
+The path to the hook executables must be absolute (e.g. ``/opt/bin/zrepl-notify.sh``).
+
+Zrepl sets a number of environment variables for the hook processes:
+
+* ``ZREPL_HOOKTYPE``: either "pre" or "post"
+* ``ZREPL_PHASE``: "Snapshotting"
+* ``ZREPL_FS``: the ZFS filesystem name being snapshotted
+* ``ZREPL_SNAPNAME``: the zrepl-generated snapshot name (e.g. ``zrepl_20380119_031407_000``)
+* ``ZREPL_DRYRUN``: indicates whether a dry run is in progress so scripts can skip some snapshot tasks
+
+No arguments may be specified; create a wrapper script if zrepl must call an executable that requires arguments.
+Each hook configuration may specify a ``pre`` script only, a ``post`` script only, or both. Each script can include an optional ``filesystems`` filter, limiting which filesystems the hook runs for. This uses the same |filter-spec| as jobs.
+
+The ``timeout`` parameter is optional and specifies a period after which zrepl will kill the hook process and report an error.
+The default is 30 seconds and may be specified in any units understood by `time.ParseDuration <https://golang.org/pkg/time/#ParseDuration>`_.
+
+The ``stop_after_error`` parameter specifies whether the snapshot process will stop after a failed hook. The default
```

Comment:
I think you found an even better name for this flag in the implementation: `error_is_fatal`